mesonbuild/dependencies/base.py

```diff
@@ -46,6 +46,17 @@ class DependencyException(MesonException):
     '''Exceptions raised while trying to find dependencies'''
 
 
+class MissingCompiler:
+    """Represent a None Compiler - when no tool chain is found.
+    replacing AttributeError with DependencyException"""
+
+    def __getattr__(self, item):
+        raise DependencyException('no toolchain found')
```

Comment:
This error message is sort of sparse on details. However, it's very difficult for it to know the name of the dependency that has the issue, so it's complicated to get more details anyway.

I do think I've found a neat solution, and I've submitted it at #10756

In the current git master, dependencies error out with terrible, confusing tracebacks.

With your PR, dependencies error out with:

```
ERROR: no toolchain found
```

With both of our PRs together, it would instead error out with e.g.

```
ERROR: Dependency lookup for zlib with method 'system' failed: no toolchain found
```

Reply:
My PR is now merged.